tests/webpay/plus/test_mall_transaction.py

```diff
@@ -1,82 +1,369 @@
-    #     return response
-
-    # def test_when_transaction_status(self):
-    #     response = MallTransaction.status(token=self.token_mock)
-    #     self.assertIsNotNone(response.vci)
-    #     self.assertIsNotNone(response.details[0].amount)
-    #     self.assertIsNotNone(response.details[0].status)
-    #     self.assertIsNotNone(response.details[0].authorization_code)
-    #     self.assertIsNotNone(response.details[0].response_code)
-    #     self.assertIsNotNone(response.details[0].installments_number)
-    #     self.assertIsNotNone(response.details[0].commerce_code)
-    #     self.assertIsNotNone(response.details[1].amount)
-    #     self.assertIsNotNone(response.details[1].status)
-    #     self.assertIsNotNone(response.details[1].authorization_code)
-    #     self.assertIsNotNone(response.details[1].response_code)
-    #     self.assertIsNotNone(response.details[1].installments_number)
-    #     self.assertIsNotNone(response.details[1].commerce_code)
+    @patch('transbank.common.request_service.requests.post')
+    def test_create_mall_transaction_successful(self, mock_post):
+        self.mock_response.status_code = 200
+        self.mock_response.text = json.dumps(responses['create_response'])
+        mock_post.return_value = self.mock_response
+
+        response = self.transaction.create(self.mall_buy_order_mock, self.session_id_mock, self.return_url_mock,
+                                           self.get_mall_transaction_details())
+
+        self.assertEqual(response, responses['create_response'])
+
+    @patch('transbank.common.request_service.requests.post')
+    def test_create_mall_exception_not_authorized(self, mock_post):
+        self.mock_response.status_code = 401
+        self.mock_response.text = json.dumps(responses['create_error'])
+        mock_post.return_value = self.mock_response
+
+        with self.assertRaises(TransactionCreateError) as context:
+            self.transaction.create(self.mall_buy_order_mock, self.session_id_mock, self.return_url_mock,
+                                    self.get_mall_transaction_details())
+
+        self.assertTrue('Not Authorized' in context.exception.message)
+        self.assertEqual(context.exception.__class__, TransactionCreateError)
+
+    def test_create_mall_exception_buy_order_max_length(self):
+        with self.assertRaises(TransbankError) as context:
+            self.transaction.create(self.mall_buy_order_mock+'too_long', self.session_id_mock, self.return_url_mock,
+                                    self.get_mall_transaction_details())
+
+        self.assertTrue("'buy_order' is too long, the maximum length" in context.exception.message)
+        self.assertEqual(context.exception.__class__, TransbankError)
+
+    def test_create_mall_exception_session_id_max_length(self):
+        with self.assertRaises(TransbankError) as context:
+            self.transaction.create(self.mall_buy_order_mock, self.token_mock, self.return_url_mock,
```

Comment:
for a better understanding of test, use a descriptive argument name. In this case, you use an argument named token as session_id , and the test is related to session_id